frontends/systolic-lang/gen-systolic.py

```diff
@@ -2,53 +2,73 @@
 
 import sys
 import textwrap
-import math
 import numpy as np
-import argparse
-import json
+from futil.futil_ast import *
+from futil.futil_utils import bits_needed
 
 # Global constant for the current bitwidth.
 BITWIDTH = 32
-PE_NAME = 'mac_pe'
 # Name of the ouput array
-OUT_MEM = 'out_mem'
-PE_DEF = """
-component mac_pe(top: 32, left: 32) -> (out: 32) {
-  cells {
-    // Storage
-    acc = prim std_reg(32);
-    mul_reg = prim std_reg(32);
-    // Computation
-    add = prim std_add(32);
-    mul = prim std_mult_pipe(32);
-  }
-
-  wires {
-
-    group do_mul<"static"=4> {
-      mul.left = top;
-      mul.right = left;
-      mul.go = !mul.done ? 1'd1;
-      mul_reg.in = mul.done ? mul.out;
-      mul_reg.write_en = mul.done ? 1'd1;
-      do_mul[done] = mul_reg.done;
-    }
-
-    group do_add {
-      add.left = acc.out;
-      add.right = mul_reg.out;
-      acc.in = add.out;
-      acc.write_en = 1'd1;
-      do_add[done] = acc.done;
-    }
-
-    out = acc.out;
-  }
-
-  control {
-    seq { do_mul; do_add; }
-  }
-}"""
+OUT_MEM = CompVar('out_mem')
+PE_NAME = 'mac_pe'
+
+
+def component_PE_DEF():
```

Comment:
Let's not do this. The PE was meant to be extracted out into a separate `.futil` file that gets imported by the generated systolic array.

Reply:
How do you want to do this then? This originally was a string assigned to a variable `PE_DEF`. If you want the string form, I could do `.emit()` as well.

Reply:
How do you feel about concatenating the string for the program with the PE for now?

Reply:
To clarify, you want:
```
component mac_pe (...) -> (...) { ... }
import "..."
component main() -> () { ... }
```
?

Reply:
No, I want:
```
PE_STRING + "\n" + prog.emit()
```

Reply:
Putting the PE_STRING before the import statements wasn't liked. It is after `prog`.